// aws-ssl-profiles/passport.js
import passport from 'passport';
import { Strategy as LocalStrategy } from 'passport-local';
import mysql from 'mysql2/promise';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';
dotenv.config();

const pool = mysql.createPool({
    host: process.env.DB_HOST || 'localhost',
    user: process.env.DB_USER || 'root',
    password: process.env.DB_PASS || '',
    database: process.env.DB_NAME || 'iana_db',
    waitForConnections: true,
    connectionLimit: 10
});

passport.use(new LocalStrategy(
    { usernameField: 'email', passwordField: 'senha' },
    async (email, senha, done) => {
        try {
            if (!email || !senha) return done(null, false, { message: 'Campos obrigatórios ausentes.' });
            const [rows] = await pool.query('SELECT * FROM usuarios WHERE email = ?', [email.trim().toLowerCase()]);
            if (!rows.length) return done(null, false, { message: 'E-mail não cadastrado.' });
            const usuario = rows[0];
            if (!usuario.senha) return done(null, false, { message: 'Senha incorreta.' });
            const ok = await bcrypt.compare(senha.trim(), usuario.senha);
            if (!ok) return done(null, false, { message: 'Senha incorreta.' });
            return done(null, usuario);
        } catch (err) { return done(err); }
    }
));

passport.serializeUser((user, done) => done(null, user.id));
passport.deserializeUser(async (id, done) => {
    try {
        const [rows] = await pool.query('SELECT id, nome, email FROM usuarios WHERE id = ?', [id]);
        done(null, rows[0] || false);
    } catch (err) { done(err); }
});

export default passport;
